contracts/L2_TokenRetriever.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import { Lockable } from "./Lockable.sol";
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { WithdrawalAdapter } from "./chain-adapters/l2/WithdrawalAdapter.sol";
+
+interface IBridgeAdapter {
+    function withdrawTokens(WithdrawalAdapter.WithdrawalInformation[] memory) external;
+}
+
+/**
+ * @notice Contract deployed on an arbitrary L2 to act as an intermediate contract for withdrawals from L3 to L1.
+ * @dev Since each network has its own bridging requirements, this contract delegates that logic to a bridge adapter contract
+ * which performs the necessary withdraw action.
+ */
+contract L2_TokenRetriever is Lockable {
```

Comment:
Can we just rename this `L2TokenRetriever`, typically there is no need for the underscore but we have made a bad pattern of this with the L1 adapters